### DISCUSSION TITLE
Recommended way of serializing HeteroData instance

### DISCUSSION THREAD

**Discussion by ghost** [pyg-team/pytorch_geometric#9147] [Q&A]
Are there any recommended ways of serializing a HeteroData instance? The use case here is that we have a featurization pipeline that produces HeteroData, and we want to store it on Google Cloud Storage for use in a training pipeline. We're currently just saving things as bytestrings but I'm wondering if there are any official ways of doing it.

**Comment by rusty1s** [comment]
If you want to be PyG version agnostic, it is recommended to save data instances as dictionary, e.g.:
```python
torch.save(data.to_dict(), path)
HeteroData.from_dict(torch.load(path))
```
I think using pickling is the easiest and best approach to do it, but there certainly exists other ways, e.g., via databases, text, csv, etc.
